### PR TITLE
feat: custom home footer

### PR DESCRIPTION
**Pull request by childrentime**
## Summary

closes: #1237 

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

**Comment by childrentime** [comment]
fix https://github.com/web-infra-dev/rspress/issues/1237

**Comment by Timeless0911** [comment]
LGTM.

You can delete `e2e/fixtures/custom-home-footer/styles/index.css` and remove line 6-15 in `e2e/fixtures/custom-home-footer/rspress.config.ts` to make test case more simple.

Can you also write some descriptions to both zh and cn doc about [footer](https://rspress.dev/zh/api/config/config-theme#footer) to introduce this feature?

File: packages/document/docs/en/api/config/config-theme.mdx [hunk 1]
```diff
@@ -209,7 +211,8 @@ import { defineConfig } from 'rspress/config';
 export default defineConfig({
   themeConfig: {
     footer: {
-      message: 'This is a footer',
+      message:
+        '<p>这是一个包含<a href="https://example.com">链接</a>的<strong>页脚</strong></p>',
```
Comment (Timeless0911) [inline packages/document/docs/en/api/config/config-theme.mdx:215]: should use english here~

**Comment by childrentime** [comment]
Yes, Minister!🤪

**Review by Timeless0911 — APPROVED** [review]

**Comment by Timeless0911** [comment]
This PR will be merged after all CI complete, and you can use nightly version tomorrow. We will not do latest release recently since we are using 1.0.0-alpha version of Rsbuild and Rspack , we want to do a minor release later.

**Pull request merged**